### DISCUSSION TITLE
Best way to include secret API keys

### DISCUSSION THREAD

**Discussion by bonzairob** [pyinstaller/pyinstaller#6973] [Help]
I'd like to add Discord rich presence to my app, using pypresence. I bundle the app in one file mode for Windows and Mac. It only requires the Discord app's ID, so it's not some super private key, but what's the best way to include something like that?

If I include like a config.py file, it will get compiled/pyc'd with the rest of the files - would that be enough to stop casual discovery?

If not, as a feature suggestion, it'd be nice if we could include simple variables on build, is that a thing?

**Comment by rokm** [comment]
> If I include like a config.py file, it will get compiled/pyc'd with the rest of the files

Yes, it would be collected as byte-compiled pyc.

> would that be enough to stop casual discovery?

Probably not. It really depends on how (un)motivated your "casual" user is. 

But it is pretty trivial to unpack PyInstaller archives with something like https://github.com/extremecoders-re/pyinstxtractor, or even our own [pyi-archive_viewer](https://github.com/pyinstaller/pyinstaller/blob/develop/PyInstaller/utils/cliutils/archive_viewer.py).

> If not, as a feature suggestion, it'd be nice if we could include simple variables on build, is that a thing?

That would not solve the problem, really - those variables would have to be stored somewhere, and it would be pretty trivial to extend tools like pyinstxtractor to extract them. Especially since PyInstaller's internals format, while perhaps not (best) documented, is not exactly a secret, either.

**Comment by bwoodsend** [comment]
In order for your application to be able to run, it has to contain enough information to get back more or less to your original code. No matter how much obfuscation you cram in, PyInstaller will never be able to do anything more than inconvenience someone trying to reverse engineer your application.
